docs/excel/custom-functions-get-started.md

```diff
@@ -0,0 +1,112 @@
+
+#### Option 1: Less nesting
+
+```js
+    /**
+    * Returns the sum of input numbers.
+    * @customfunction
+    */
+    function Add(args: number[]): number {
+      let total = 0;
+      args.forEach(value => {
+        total += value;
+      });
+     
+      return total;
+    }
+```
+
+#### Option 2: More nesting
+
+```js
+    /**
+    * Returns the sum of two numbers.
+    * @customfunction
+    */
+    function Add(arg1: number, arg2: number): number {
+      return arg1 + arg2;
+    }
+```
+
+### Excel calculation mode
+
+Excel has three calculation modes: Automatic, Automatic Except Tables, and Manual. For a description of these calculation modes, see [Excel Recalculation](/office/client-developer/excel/excel-recalculation). The most frequently used calculation mode for custom functions add-ins is manual calculation mode. Set the calculation mode for your add-in with the [Excel.CalculationMode enum](/javascript/api/excel/excel.calculationmode) based on your scenario. Note that automatic calculation mode may trigger recalculation often and reduce the efficiency of your add-in.
+
+### Volatile function limitations
+
+Custom functions allow you to create your own volatile functions, similar to the `NOW` and `TODAY` functions in Excel. During recalculation, Excel evaluates cells that contain volatile functions and all of their dependent cells. As a result, using many volatile functions may make recalculation slow. Volatile functions should be used sparingly to optimize your custom functions add-in. For additional information, see [Volatile and Non-Volatile Functions](/office/client-developer/excel/excel-recalculation#volatile-and-non-volatile-functions).
+
+## Design approaches to improve efficiency
+
+Custom functions add-ins allow for flexible designs, which means that different add-in designs can produce the same output for your end users.
+
+### Multiple results
+
+You can return multiple results from your custom function with multiple functions or with one function.
+
+To return multiple results with one function, use a dynamic array. This is usually the recommended approach because dynamic arrays only require updating a single cell to trigger recalculation for all results. To learn more about dynamic arrays in custom functions, see [Return multiple results from your custom function](custom-functions-dynamic-arrays.md).
+
+Another way to return multiple results is to use multiple functions and return a single result for each function. A benefit of using multiple functions is that your end user can decide precisely which formula they want to update and then only trigger recalculation for that formula.
```

Comment:
Might be good to show example like in the spec. Also point out the tradeoff of how one function for multiple results is less efficient across a larger dataset. Especially if relying on an external service.